pypesto/optimize/ess/cess.py

```diff
@@ -0,0 +1,340 @@
+"""Cooperative Enhanced Scatter Search."""
+import logging
+import multiprocessing
+import os
+import time
+from typing import Dict, List, Optional
+
+import numpy as np
+
+import pypesto.optimize
+from pypesto import Problem
+from pypesto.startpoint import StartpointMethod
+
+from .ess import ESSExitFlag, ESSOptimizer
+from .function_evaluator import FunctionEvaluator
+from .refset import RefSet
+
+logger = logging.getLogger(__name__)
+
+
+class CESSOptimizer:
+    r"""
+    Cooperative Enhanced Scatter Search Optimizer (CESS).
+
+    A cooperative scatter search algorithm based on [VillaverdeEge2012]_.
+    In short, multiple scatter search instances with different hyperparameters
+    are running in different threads/processes, and exchange information.
+    Some instances focus on diversification while others focus on
+    intensification. Communication happens at fixed time intervals.
+
+    Proposed hyperparameter values in [VillaverdeEge2012]_:
+
+    * ``dim_refset``: ``[0.5 n_parameter, 20 n_parameters]``
+    * ``local_n2``: ``[0, 100]``
+    * ``balance``: ``[0, 0.5]``
+    * ``n_diverse``: ``[5 n_par, 20 n_par]``
+    * ``max_eval``: such that :math:`\tau = log10(max_eval / n_par)` is in
+      [2.5, 3.5], with a recommended default value of 2.5.
+
+    .. [VillaverdeEge2012] 'A cooperative strategy for parameter estimation in
+       large scale systems biology models', Villaverde, A.F., Egea,
+       J.A. & Banga, J.R. BMC Syst Biol 2012, 6, 75.
+       https://doi.org/10.1186/1752-0509-6-75
+
+    Attributes
+    ----------
+    ess_init_args:
```

Comment:
duplicate of init docstring?

Reply:
Class attributes vs. __init__ arguments. But, yes, significant overlap.